src/features/island/plots/Plot.tsx

```diff
@@ -136,21 +132,21 @@ export const Plot: React.FC<Props> = ({ id }) => {
       return;
     }
 
+    // harvest crop when ready
+    if (readyToHarvest) {
+      harvestCrop(crop);
+    }
+
     // apply fertilisers
-    if (selectedItem && selectedItem in FERTILISERS) {
+    if (!readyToHarvest && selectedItem && selectedItem in CROP_COMPOST) {
```

Comment:
Either this should be "else-connected" to L136 (for clarity) or `readyToHarvest` should be updated to reflect the impact of L137.